crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1716,6 +1717,154 @@ unsafe impl<'__w, T: Component> QueryData for Mut<'__w, T> {
     type ReadOnly = Ref<'__w, T>;
 }
 
+/// A collection of potentially conflicting [`QueryData`]s allowed by disjoint access.
```

Comment:
The docs should make it clear that the query will fetch entitieis that match the union of all the QueryData's in the DataSet.